pkg/apis/eventsource/v1alpha1/types.go

```diff
@@ -288,6 +290,31 @@ type FileEventSource struct {
 	Filter *EventSourceFilter `json:"filter,omitempty" protobuf:"bytes,5,opt,name=filter"`
 }
 
+// SFTPEventSource describes an event-source for sftp related events.
+type SFTPEventSource struct {
+	// Type of file operations to watch
+	// Refer https://github.com/fsnotify/fsnotify/blob/master/fsnotify.go for more information
+	EventType string `json:"eventType" protobuf:"bytes,1,opt,name=eventType"`
+	// WatchPathConfig contains configuration about the file path to watch
+	WatchPathConfig WatchPathConfig `json:"watchPathConfig" protobuf:"bytes,2,opt,name=watchPathConfig"`
+	// Username required for authentication if any.
+	Username *corev1.SecretKeySelector `json:"username,omitempty" protobuf:"bytes,3,opt,name=username"`
+	// Password required for authentication if any.
+	Password *corev1.SecretKeySelector `json:"password,omitempty" protobuf:"bytes,4,opt,name=password"`
+	// Address sftp address.
+	Address *corev1.SecretKeySelector `json:"address,omitempty" protobuf:"bytes,5,opt,name=address"`
+	// Metadata holds the user defined metadata which will passed along the event payload.
+	// +optional
+	Metadata map[string]string `json:"metadata,omitempty" protobuf:"bytes,6,rep,name=metadata"`
+	// Filter
+	// +optional
+	Filter *EventSourceFilter `json:"filter,omitempty" protobuf:"bytes,7,opt,name=filter"`
+	// PollIntervalDuration the interval at which to poll the SFTP server
+	// defaults to 10 seconds
+	// +optional
+	PollIntervalDuration string `json:"pollIntervalDuration" protobuf:"varint,8,opt,name=pollIntervalDuration"`
```

Comment:
Use `*string` to avoid parsing empty string?

Reply:
Is that preferred? I'm following the pattern defined by `HDFSEventSource` with `CheckInterval`

https://github.com/dillonstreator/argo-events/blob/980c63b8576969e67f27e91c65e049b5f9d6e072/pkg/apis/eventsource/v1alpha1/types.go#L1091-L1093
https://github.com/dillonstreator/argo-events/blob/980c63b8576969e67f27e91c65e049b5f9d6e072/eventsources/sources/hdfs/start.go#L96-L104